### PR TITLE
Bugfix: Resymmetrize logical coordinates

### PR DESCRIPTION
**Pull request by lroberts36**
## PR Summary

Originally, Athena++ transformed integer logical coordinates into floating point logical coordinates in the range [-0.5, 0.5]. In #911, I changed this to map to the range [0, 1]. As @pgrete has pointed out, the new range breaks floating point symmetry (see the comment at the end of this athena++ [document](https://github.com/PrincetonUniversity/athena/wiki/Coordinate-Systems-and-Meshes#tips-for-athena-developers) at the very bottom).  This in turn causes static mesh refinement to operate differently due to roundoff error and block coordinates to not be bitwise exact with the original coordinate positions for single tree forests. This PR switches back to symmetrized floating-point logical coordinates and back to the exact algorithm used for finding static refinement regions from before #911.  

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [x] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by pdmullen — APPROVED** [review]
Round-off error strikes again! Happy to approve assuming that it fixes @pgrete's issue.  Also, I'd like to understand why this only affects SMR? Shouldn't this have affected symmetry more generally?

**Comment by lroberts36** [comment]
It also effects the coordinates of blocks at roundoff, but I don't think this is as big of an issue as the impact on SMR (where the refinement structure was different in some cases with the exact same input). That being said, it probably broke exact symmetry (if it ever existed in any downstream code).

**Review by Yurlungur — APPROVED** [review]
LGTM. I also like the new names better.

File: src/mesh/mesh.cpp [hunk 1]
```diff
@@ -1065,22 +1065,26 @@ void Mesh::DoStaticRefinement(ParameterInput *pin) {
...
+    int lxmin, lxmax;
+    for (lxmin = 0; lxmin < lxtot; lxmin++) {
+      Real r = LogicalLocation::IndexToSymmetrizedCoordinate(lxmin + 1,
+                                                             BlockLocation::Left, lxtot);
+      if (mesh_size.SymmetrizedLogicalToActualPosition(r, dir) > ref_size.xmin(dir))
+        break;
+    }
+    for (lxmax = lxmin; lxmax < lxtot; lxmax++) {
+      Real r = LogicalLocation::IndexToSymmetrizedCoordinate(lxmax + 1,
+                                                             BlockLocation::Left, lxtot);
```
Comment (pgrete) [inline src/mesh/mesh.cpp:1081]: Just double checking, this is not meant to be `BlockLocation::Right` is it? (the new logical location machinery is still a little opaque to me).

**Review by pgrete — APPROVED** [review]
LGTM and I can confirm that this results in the expected mesh again (though other issue still persist with restarts, but I'll track those down separately).

I suggest to wait clicking merge on this PR until #1055 is in `develop`, then change the target branch and pull the trigger here.
This way we get a clean history (which may at some point be useful for bisects)

File: src/mesh/mesh.cpp (hunk 1 above)
Comment (lroberts36) [reply src/mesh/mesh.cpp:1081]: It is supposed to be `BlockLocation::Left` here. I agree this is a little confusing, but this gives code that is as close as possible to the old implementation.

**Pull request merged**